vortex-buffer/src/buffer.rs

```diff
@@ -177,8 +177,10 @@ impl<T> Buffer<T> {
     }
 
     /// Returns an iterator over the buffer of elements of type T.
-    pub fn iter(&self) -> impl Iterator<Item = &T> + '_ {
-        self.as_slice().iter()
+    pub fn iter(&self) -> Iter<'_, T> {
```

Comment:
I think this technically isn't a breaking change actually, b/c we're going from unnamed -> named type.

But the idea is that we have to impl `TrustedLen` for this, so you can BufferMut::extend_trusted from another Buffer.

So I just defined a concrete type that wraps std::slice::Iter and delegates the important methods to it.